File: libraryCoreJava/src/main/java/com/themone/core/base/impl/BaseModel.java
package com.themone.core.base.impl;

import androidx.annotation.NonNull;

import com.themone.core.base.IBaseModel;
import com.themone.core.http.HttpClient;
import com.themone.core.util.LogUtil;

/**
 * @author zhiqiang
 * @date 2019-06-04
 * @desc
 */
public class BaseModel<T> implements IBaseModel {

    /**
     * apiService
     * 用于 retrofit 请求网络
     */
    protected T mApiService;

    public BaseModel(@NonNull Class<T> clazz) {
        this.mApiService = HttpClient.create(clazz);
    }

    @Override
    public void onDestroy() {
        LogUtil.i("BaseModel", "onDestroy");
        mApiService = null;
    }

}

File: libraryframework/src/main/java/com/theone/framework/base/BaseModel.kt
package com.theone.framework.base

import com.themone.core.base.impl.CoreModel
import com.theone.framework.http.HttpClient

/**
 * @Author zhiqiang
 * @Date 2019-08-02

 * @Description
 */
open class BaseModel<T>(clazz: Class<T>) : CoreModel<T>() {
    /**
     * apiService
     * 用于 retrofit 请求网络
     */
    protected val apiService: T = HttpClient.create(clazz)
}